artiq/coredevice/phaser.py

```diff
@@ -197,6 +200,9 @@ def init(self, debug=False):
             raise ValueError("large number of frame CRC errors")
         delay(.1*ms)  # slack
 
+        # determine the origin for frame-aligned timestamps
+        self.measure_frame_timestamp()
```

Comment:
Why here? Do it first before everything.

Reply:
I thought it should at least be after board identification. Isn't this useless to do this if the wrong board is connected or has a faulty gateware?

Reply:
Not wrong board connected (in those cases `measure_frame_timestamp()` will still work just fine), but wrong `channel_base`: `rtio_input_timestamp()` has a timeout. `rtio_input_data()` just blocks.

Reply:
With 

```python
board_id = self.read8(PHASER_ADDR_BOARD_ID)
if board_id != PHASER_BOARD_ID:
    raise ValueError("invalid board id")

self.measure_frame_timestamp()
```

and an invalid `channel_base`, `init()` fails immediately with `"invalid board id"`. Reversed, the call to `rtio_input_timestamp()` times out and then the kernel fails with `"invalid board id"`. Is there a reason to wait for the timeout in this scenario? or do I misunderstand something?

Reply:
Are you sure? On a (wrong) RTIO channel that doesn't do input events, there should never be an input event and it should block.

Reply:
Or should one check for timeout with `self.frame_tstamp == -1` and raise an exception from there (maybe `RuntimeError`)?

Reply:
Anyway, this isn't critical. Your choice were to place it.